test/general/openpgp.js

```diff
@@ -1821,6 +1846,24 @@ aOU=
       })).to.be.rejectedWith(/No signing keys provided/);
     });
 
+    it('Signing with key which uses sha3 should generate a valid sha3 signature', async function() {
+      const privKey = await openpgp.readKey({ armoredKey: priv_key_sha3 });
+      const pubKey = privKey.toPublic();
+      const text = 'Hello, world.';
+      const message = await openpgp.createCleartextMessage({ text });
+      const config = { minRSABits: 1024 };
+
+      const cleartextMessage = await openpgp.sign({ message, signingKeys: privKey, config, format: 'armored' });
+      const parsedArmored = await openpgp.readCleartextMessage({ cleartextMessage });
+      expect(parsedArmored.text).to.equal(text);
+      expect(parsedArmored.signature.packets.filterByTag(openpgp.enums.packet.signature)).to.have.length(1);
+
+      const objectMessage = await openpgp.sign({ message, signingKeys: privKey, config, format: 'object' });
+      expect(objectMessage.signature.packets.filterByTag(openpgp.enums.packet.signature)).to.have.length(1);
+      const verified = await openpgp.verify({ message: objectMessage, verificationKeys: pubKey, expectSigned: true, config });
```

Comment:
`parsedArmored` is an `CleartextMessage` object just like the one returned when requesting `format: 'object'` , and you can pass it directly to verification:

```suggestion
      const verified = await openpgp.verify({ message: parsedArmored, verificationKeys: pubKey, expectSigned: true, config });
```